src/tree/updater_quantile_hist.h

```diff
@@ -277,6 +277,68 @@ class QuantileHistMaker: public TreeUpdater {
       }
     };
 
+
+    // Drives execution of tree building on device
+    struct Driver {
```

Comment:
I can help extract the driver into a more reusable module if needed.

Reply:
@trivialfis thanks for help, but seems I already did it!
Could you check the generalization of `Driver` for cpu and gpu updaters (common header file, methods for `ExpandEntry`, ...)?

Reply:
@trivialfis  unfortunately I don't have complete understanding why previous and current commits are failed on gpu tests.
https://xgboost-ci.net/blue/organizations/jenkins/xgboost-win64/detail/PR-7007/6/pipeline
Continue to discover the exact reason of failed tests. 
without unification (462d1d3) it was ok :)

Reply:
Thanks, I will take a look tomorrow.